File: Java/problem36.java
public class Solution 
{
    public boolean isValidSudoku(char[][] board)
    {
        for(int i = 0; i < 9; ++i)
        {
            int[] mark = {0, 0, 0, 0, 0, 0, 0, 0, 0};
            for(int j = 0; j < 9; ++j)
            {
                if(board[i][j] != '.')                  //先对每一行作判断
                    if(++mark[(int)(board[i][j]) - 49] == 2)
                        return false;
            }
            int[] markCol = {0, 0, 0, 0, 0, 0, 0, 0, 0};
            for(int j = 0; j < 9; ++j)
            {
                if(board[j][i] != '.')                  //再对每一列作判断
                    if(++markCol[(int)(board[j][i]) - 49] == 2)
                        return false;
            }
        }
        for(int i = 0; i < 3; ++i)                      //对每一个方格作判断
        {
            for(int j = 0; j < 3; ++j)
            {
                int[] mark = {0, 0, 0, 0, 0, 0, 0, 0, 0};
                for(int k = 0; k < 3; ++k)
                {
                    for(int m = 0; m < 3; ++m)
                    {
                        if(board[i*3 + k][j*3 + m] != '.')
                            if(++mark[(int)(board[i*3 + k][j*3 + m]) - 49] == 2)
                                return false;
                    }
                }
            }
        }
        return true;
    }
}

File: python/problem36.py
class Solution(object):
    def isValidSudoku(self, board):
        """
        :type board: List[List[str]]
        :rtype: bool
        """
        for i in range(9):
            mark = [0] * 9;
            for j in range(9):
                if board[i][j] != '.':                  #先对每一行作判断
                    mark[(int)(board[i][j])-1] += 1;
                    if mark[(int)(board[i][j])-1] == 2:
                        return False;
            
            markCol = [0] * 9;
            for j in range(9):
                if board[j][i] != '.':                  #再对每一列作判断
                    markCol[(int)(board[j][i])-1] += 1;
                    if markCol[(int)(board[j][i])-1] == 2:
                        return False;
        
        for i in range(3):                              #对每一个方格作判断
            for j in range(3):
                mark = [0] * 9;
                for k in range(3):
                    for m in range(3):
                        if board[i*3 + k][j*3 + m] != '.':
                            mark[(int)(board[i*3 + k][j*3 + m])-1] += 1;
                            if mark[(int)(board[i*3 + k][j*3 + m])-1] == 2:
                                return False;
        return True;